### PR TITLE
ht19 - initial commit

### PR DESCRIPTION
**Pull request by CherpakovPhilipp**

### CODE REVIEW

File: src/components/simpleInput/Input.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+class Input extends Component {
+  state = {
+    inputText: '',
+    callback: null
+  };
+
+  changeField = (event) => {
+    this.setState({ inputText: event.target.value });
+    this.props.setInputText(this.state.inputText);
```
Comment (iliatcymbal) [inline src/components/simpleInput/Input.js]: 1) why do you save input text in the state here and in the state of `SimpleInput` component?
2) setState is async function, so you pass to `this.props.setInputText` an obsolete value, it doesn't work as expected

Have you checked how your component works? Check this short [video](https://www.dropbox.com/s/4mf37fov1uapxnv/inputIssue.gif?dl=0) to understand what I'm talking about

File: src/components/simpleInput/Input.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  render() {
+    const { inputText, hidden } = this.state;
+
+    return (
+      <form
+        className="input-block"
+        onClick={this.onClick}
+        onBlur={this.onBlur}
```
Comment (iliatcymbal) [inline src/components/simpleInput/Input.js]: onBlur should be set on input

File: src/components/simpleInput/Input.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      >
+        <Input
+          className={hidden ? 'hide' : 'show'}
+          setInputText={this.setInputText}
+          callback={this.props.callback}
+        />
+        <Span
+          className={`${hidden ? 'show' : 'hide'} output`}
+          text={inputText}
+        />
```
Comment (iliatcymbal) [inline src/components/simpleInput/Input.js]: Use simple `<span...` and `<input...` here, there are no any reasons to split this component. It is really hard to understand what's going on here

File: src/components/simpleInput/Input.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import './input.scss';
+
```
Comment (iliatcymbal) [inline src/components/simpleInput/Input.js]: Why is file called `Input.js` if it exports `SimpleInput`?

File: src/components/infoBar/InfoBar.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+export const Infobar = props => (
+  <div className="info-bar">
+    <h2>
+Hello,
```
[the remark this answers is not in the document]
Comment (CherpakovPhilipp) [reply src/components/infoBar/InfoBar.js]: this is because of eslint)

File: src/components/simpleInput/SimpleInput.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        className="input-block"
+        onClick={this.onClick}
+      >
+        <h2>input/output</h2>
+        <input
+          type="text"
+          value={this.state.inputText}
+          onChange={this.changeField}
+          onBlur={this.onBlur}
+          className={hidden ? 'hide' : 'show'}
```
Comment (iliatcymbal) [inline src/components/simpleInput/SimpleInput.js]: You have to show and hide components using conditional rendering, i.e. `true ? <span /> : <input />`

**Pull request merged**